lib/core/src/Cardano/Wallet/Address/Pool.hs

```diff
@@ -136,10 +138,21 @@ prop_fromIx Pool{addressFromIx,addresses} =
   where
     isGenerated addr (ix,_) = addressFromIx ix == addr
 
+prop_next_is_unused :: (Eq ix, Enum ix) => Pool addr ix -> Bool
+prop_next_is_unused p =
+    let i = nextIndex p
+        indexState = filter ((== i) . fst) (Map.elems (addresses p))
+    in  indexState == [(i, Unused)]
+
 -- | Internal invariant: The pool satisfies all invariants above.
 prop_consistent :: (Ord ix, Enum ix, Eq addr) => Pool addr ix -> Bool
-prop_consistent p =
-    all ($ p) [prop_sequence, prop_gap, prop_fresh, prop_fromIx]
+prop_consistent p = all ($ p)
+    [ prop_sequence
+    , prop_gap
+    , prop_fresh
+    , prop_fromIx
+    , prop_next_is_unused
```

Comment:
```suggestion
```

This property is actually a consequence of the other properties, so it's not required for consistency.